xarray/core/variable.py

```diff
@@ -2072,8 +2072,11 @@ def _coarsen_reshape(self, windows, boundary, side):
             else:
                 shape.append(variable.shape[i])
 
-        keep_attrs = _get_keep_attrs(default=False)
```

Comment:
I think we want this one level up in `.coarsen`. This function returns `array_like, tuple`. The preservation of attrs is a confusing side effect here.

Reply:
I left it at the end of _coarsen_reshape because that is where the code is currently:

```
keep_attrs = _get_keep_attrs(default=False)
variable.attrs = variable._attrs if keep_attrs else {}
```

it just didn't seem to be working as intended.

Reply:
Thanks. This may be why it's changing the attrs of the original variable. Passing a new `attrs` to `._replace` above should be clearer and easier to understand.

Reply:
I just pushed changes to use this method.